### PR TITLE
Remove compilation bottleneck for Swift 5.3.2 builds

### PR DESCRIPTION
**Pull request by iainsmith**
### Description
Improve compile times for consumers using older versions of Xcode/swift. This change significantly improves compile times when using Swift 5.3.2. More details & benchmarks in #146 

*Edit* - Pushed a small change to the wording of the comment

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

File: Sources/Algorithms/Stride.swift [hunk 1]
```diff
@@ -226,9 +226,15 @@ extension StrideCollection: Collection {
     offsetBy n: Int,
     limitedBy limit: Index
   ) -> Index? {
-    let distance = i == endIndex
-      ? -((base.count - 1) % stride + 1) + (n - 1) * -stride
-      : n * -stride
+    // We typically use the ternary operator but this significantly increases compile
+    // times when using Swift 5.3.2
```
Comment (xwu) [inline Sources/Algorithms/Stride.swift:230]: Small formatting nit:

```suggestion
    // We typically use the ternary operator but this significantly increases
    // compile times when using Swift 5.3.2
```

**Review by xwu — COMMENTED** [review]
I'd be curious to see if compile times can be addressed on Swift 5.3.2 by replacing each occurrence of literal `1` with `(1 as Int)`.

File: Sources/Algorithms/Stride.swift (hunk 1 above)
Comment (iainsmith) [reply Sources/Algorithms/Stride.swift]: Thanks @xwu, I've merged it with a note to squash it into the other commit.

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test

**Comment by iainsmith** [comment]
Looks like the Mac agent failed to clone the repo

'''
git fetch --tags --force --progress -- git@github.com:apple/swift-algorithms.git +refs/heads/*:refs/remotes/origin/* # timeout=10
07:07:37 ERROR: Error cloning remote repo 'origin'
07:07:37 hudson.plugins.git.GitException: Command "git fetch --tags --force --progress -- 
'''

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**